pkg/networkservice/chains/endpoint/server.go

```diff
@@ -54,28 +56,69 @@ type endpoint struct {
 	networkservice.MonitorConnectionServer
 }
 
+type serverOptions struct {
+	name                    string
+	authzServer             networkservice.NetworkServiceServer
+	tokenGenerator          token.GeneratorFunc
+	additionalFunctionality []networkservice.NetworkServiceServer
+}
+
+// Option modifies server option value
+type Option func(o *serverOptions)
+
+// WithName sets name of the NetworkServiceServer
+func WithName(name string) Option {
+	return func(o *serverOptions) {
+		o.name = name
+	}
+}
+
+// WithAuthorizationServer sets authorization server chain element
+func WithAuthorizationServer(authzServer networkservice.NetworkServiceServer) Option {
+	return func(o *serverOptions) {
+		o.authzServer = authzServer
+	}
+}
+
+// WithTokenGenerator sets token generator function - generates tokens for use in Path
+func WithTokenGenerator(tokenGenerator token.GeneratorFunc) Option {
+	return func(o *serverOptions) {
+		o.tokenGenerator = tokenGenerator
+	}
+}
```

Comment:
This is the required parameter for the endpoint...

Reply:
removed from options